node/builder.go

```diff
@@ -341,6 +341,38 @@ func Repo(r repo.Repo) Option {
 	}
 }
 
+func Provider(r repo.Repo) Option {
+	return func(settings *Settings) error {
+		lr, err := r.Lock(settings.nodeType)
+		if err != nil {
+			return err
+		}
+		c, err := lr.Config()
+		if err != nil {
+			return err
+		}
+		_ = c
+		return Options(
+			func(s *Settings) error { s.Base = true; return nil }, // mark Base as applied
+			ApplyIf(func(s *Settings) bool { return s.Config },
+				Error(errors.New("the Base() option must be set before Config option")),
+			),
+			Override(new(repo.LockedRepo), modules.LockedRepo(lr)), // module handles closing
+
+			Override(new(ci.PrivKey), lp2p.PrivKey),
+			Override(new(ci.PubKey), ci.PrivKey.GetPublic),
+			Override(new(peer.ID), peer.IDFromPublicKey),
```

Comment:
Shouldn't need this, it's just libp2p stuff, which we don't want on lotus-provider